pkg/proxy/topology_test.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"testing"
+)
+
+func TestFilterTopologyEndpoint(t *testing.T) {
+	testCases := []struct {
+		nodeMap         NodeMap
+		endpoints       []Endpoint
+		currentNodeName types.NodeName
+		topologyKeys    []string
+		expected        []Endpoint
+	}{
+		{
+			// Case[0]: no endpoint
+			nodeMap: NodeMap{
+				"testNode1": &BaseNodeInfo{
+					name: "testNode",
+					labels: map[string]string{
+						"kubernetes.io/hostname":                   "10.0.0.1",
+						"failure-domain.beta.kubernetes.io/zone":   "90001",
+						"failure-domain.beta.kubernetes.io/region": "sg",
+					},
+				},
+			},
+			endpoints:       []Endpoint{},
+			currentNodeName: "testNode",
+			topologyKeys:    []string{"failure-domain.beta.kubernetes.io/region"},
+			expected:        []Endpoint{},
+		},
+		{
+			// Case[1]: no topologyKeys
```

Comment:
These would be easier to read with some helpers, e.g.:

```
nodeMap: NodeMap{
    "testNode1": makeNode("testNode1", map[string]string{
        "kubernetes.io/hostname":                 "10.0.0.1",
￼		"failure-domain.beta.kubernetes.io/zone": "90001",
       })
    "testNode2": makeNode("testNode2", nil)
}, 
endpoints: []Endpoint{
    makeEndpoint("1.1.1.1:11", "testNode1"),
    makeEndpoint("1.1.1.2:22", "testNode2"),
}
currentNodeName: "testNode1",
topologyKeys: nil,
expected: []Endpoint{
    makeEndpoint("1.1.1.1:11", "testNode1"),
    makeEndpoint("1.1.1.2:22", "testNode2"),
}
```

Reply:
done